pkg/core/server/server.go

```diff
@@ -44,22 +46,25 @@ func NewServer(
 // Handle delegates gorilla mux Handler, and accept a HandlerFactory instead of Handler
 func (s *Server) Handle(path string, hf handler.Factory) *mux.Route {
 	return s.router.NewRoute().Path(path).Handler(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		r = auth.InitRequestAuthContext(r)
+		r = db.InitRequestDBContext(r)
```

Comment:
This init is required?

How about if a handler is not related to an DB operation.

Reply:
This init does really interact with the DB,

It can be removed actually if everything is lazily created.

Reply:
Oh, sorry, this init is required.

Because context in request is immutable, creating the container lazily would not work.